From 2eddec8a81ae9ceedd8246fbcd084f0cfa2ab7eb Mon Sep 17 00:00:00 2001
From: toge <toge.mail@gmail.com>
Date: Mon, 1 Jul 2024 22:50:38 +0900
Subject: [PATCH] Include algorithm header for gcc 14. (#5139)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

<long description>

There is a compilation error on gcc 14.
```
tiledb/sm/filesystem/posix.cc: In static member function ‘static void tiledb::sm::Posix::adjacent_slashes_dedup(std::string*)’:
tiledb/sm/filesystem/posix.cc:384:12: error: ‘unique’ is not a member of ‘std’
  384 |       std::unique(
      |            ^~~~~~
```
This patch will fix this issue.

---
TYPE: NO_HISTORY
DESC: Include <algorithm> header for gcc 14.

---------

Co-authored-by: Theodore Tsirpanis <theodore.tsirpanis@tiledb.com>
---
 tiledb/sm/filesystem/posix.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tiledb/sm/filesystem/posix.cc b/tiledb/sm/filesystem/posix.cc
index 6e0b678941b..a4419338be2 100644
--- tiledb/sm/filesystem/posix.cc
+++ tiledb/sm/filesystem/posix.cc
@@ -47,6 +47,7 @@
 #include <sys/stat.h>
 #include <unistd.h>
 
+#include <algorithm>
 #include <fstream>
 #include <future>
 #include <iostream>
